### DISCUSSION TITLE
Help request: Dealing with "The spec started, but never completed" occurring mid run - any way to continue to the next test/spec?

### DISCUSSION THREAD

**Discussion by ghowardzoonou** [cypress-io/cypress#30407] [Cypress Cloud]
Hello, so in quite a large set of tests we have several spec files that we run overnight using Cypress Run, and connect the results to Cypress Cloud.

We occasionally see instances of Cypress apparently getting stuck mid-run. We're looking into root causes but that's an on-going thing as it's not a simple investigation/fix. Lot's of moving parts and no useful errors being thrown even with Cypress DEBUG logging enabled.

The main problem we have is that it seems to happen somewhat sporadically in different places during a full run. And sometimes we we'll go a few runs in a row without it occurring.

When the issue does occur - Cypress Cloud reports:  "The spec started, but never completed" and then essentially times-out the rest of the run until the pipeline side time-limit is reached and the run gets killed. So if the issue is encountered in the 2nd spec - the following 20 specs will then also fail to execute, all reporting the above error.

Does anyone know/can point to some kind of workaround/wrapper we can employ so that if Cypress gets stuck and essentially fails to complete a test scenario within x minutes, that we can then kind of force kill the current test execution and forcibly move onto the next one? Rather than it just waiting forever until the process is killed?

**Comment by ak-ameena** [comment]
@ghowardzoonou were you able to find the root cause(s) of this or develop a workaround?
